mythril/laser/ethereum/instructions.py

```diff
@@ -1057,9 +1057,45 @@ def returndatacopy_(self, global_state: GlobalState) -> List[GlobalState]:
         :param global_state:
         :return:
         """
-        # FIXME: not implemented
         state = global_state.mstate
-        start, s2, size = state.stack.pop(), state.stack.pop(), state.stack.pop()
+        memory_offset, return_offset, size = (
+            state.stack.pop(),
+            state.stack.pop(),
+            state.stack.pop(),
+        )
+        concrete_size = None
+        concrete_return_offset = None
+
+        try:
+            concrete_memory_offset = helper.get_concrete_int(memory_offset)
+        except TypeError:
+            log.debug("Unsupported symbolic memory offset in RETURNDATACOPY")
+            return [global_state]
+
+        try:
+            concrete_return_offset = helper.get_concrete_int(return_offset)
+        except TypeError:
+            log.debug("Unsupported symbolic return offset in RETURNDATACOPY")
+
+        try:
+            concrete_size = helper.get_concrete_int(size)
+        except TypeError:
+            log.debug("Unsupported symbolic max_length offset in RETURNDATACOPY")
+
+        if (
+            concrete_size is None
+            or global_state.last_return_data is None
+            or concrete_return_offset is None
+        ):
+            return [global_state]
+
+        global_state.mstate.mem_extend(concrete_memory_offset, concrete_size)
+        for i in range(concrete_size):
+            global_state.mstate.memory[concrete_memory_offset + i] = (
+                global_state.last_return_data[concrete_return_offset + i]
```

Comment:
what if there is no last return data? returndatasize does have logic handling this

Reply:
It's checked in the above if.

Reply:
Ah I missed that